lib/util/git/index.ts

```diff
@@ -568,6 +568,43 @@ export async function isBranchModified(branchName: string): Promise<boolean> {
   return true;
 }
 
+export async function isBranchConflicted(
+  baseBranch: string,
+  branch: string
+): Promise<boolean> {
+  await syncGit();
+
+  if (!branchExists(baseBranch) || !branchExists(branch)) {
+    logger.warn(
+      { baseBranch, branch },
+      'isBranchConflicted: branch does not exist'
+    );
+    return true;
+  }
+
+  let result = false;
+
+  const origBranch = config.currentBranch;
+  try {
+    if (origBranch !== baseBranch) {
+      await git.checkout(['-f', baseBranch, '--']);
+    }
+    await git.reset(ResetMode.HARD);
+    await git.merge(['--no-commit', '--no-ff', `origin/${branch}`]);
+  } catch (err) {
+    result = true;
+    // istanbul ignore if: not easily testable
+    if (!err?.git?.conflicts?.length) {
+      logger.debug(
+        { baseBranch, branch, err },
+        'Conflict detection: unknown error'
+      );
+    }
+  }
+
+  return result;
```

Comment:
I think it needs the `git merge --abort` added here (within its own try/catch maybe?

Reply:
Ah, now I see we can't rely on `syncGit()` for cleaning it up